cmake/CMakeLists.txt

```diff
@@ -310,6 +310,16 @@ if (onnxruntime_EXTENDED_MINIMAL_BUILD AND NOT onnxruntime_MINIMAL_BUILD)
   set(onnxruntime_MINIMAL_BUILD ON)
 endif()
 
+# work around Android NDK bug which doesn't include -O flag for Release configuration
+# https://github.com/android/ndk/issues/1740
+# TODO: remove this when the NDK version(s) we support get fixed
+if (CMAKE_SYSTEM_NAME STREQUAL "Android")
+  # NB: attempting to match the effects of this fix: https://android-review.googlesource.com/c/platform/ndk/+/2168845
+  string(APPEND CMAKE_C_FLAGS_RELEASE " -O3")
```

Comment:
CMAKE has 4 build types. Your this change only impacts one of them: "release". So it may not be enough. 

You can also build it in a way like:

https://github.com/microsoft/onnxruntime/blob/master/tools/ci_build/github/azure-pipelines/templates/c-api-linux-cpu.yml#L61

```
docker run --rm -e CFLAGS="${{parameters.OnnxruntimeCFlags}}" -e CXXFLAGS="${{parameters.OnnxruntimeCXXFlags}}"
```
In this way it will override the default values.

Reply:
"Release" is the one we want to fix, so it is the only one that is updated here.

My understanding is that `CXXFLAGS` will initialize [`CMAKE_CXX_FLAGS`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS.html) and [`CMAKE_CXX_FLAGS_RELEASE`](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS_CONFIG.html#variable:CMAKE_%3CLANG%3E_FLAGS_%3CCONFIG%3E) values are passed after `CMAKE_CXX_FLAGS`.